docs/logging.md

```diff
@@ -22,3 +77,92 @@ code, time to process, (optional perf data details, e.g. database time/queries,
 memcached time/queries, Django process startup time, markdown processing time,
 etc.), URL, and "email via client" showing user account involved (if logged in)
 and the type of client they used ("web", "Android", etc.).
+
+The performance data details are particularly useful for investigating
+performance problems, since one can see at a glance whether a slow
+request was caused by delays in the database, in the markdown
+processor, in memcached, or in other Python code.
+
+One useful thing to note, however, is that the database time is only
+the time spent connecting to and receiving a response from the
+database.  Especially when response are large, there can often be a
+great deal of Python processing overhead to marshall the data from the
+database into Django objects that is not accounted for in these
+numbers.
+
+## Blueslip frontend error reporting
+
+We have a custom library, called `blueslip` (named after the form used
+at MIT to report problems with the facilities), that takes care of
+reporting JavaScript errors.  In production, this means emailing the
+server administrators (though the setting controlling this,
+`BROWSER_ERROR_REPORTING`, is disabled by default, since JavaScript
+errors tend to have fairly high volume and are almost never
+configuration problems).  In development, this means displaying a
```

Comment:
I'd say what kind of problems they are, since otherwise when not reading closely it sounds a bit like dismissing these errors. I think the point is that they're either bugs in our code, or in the user's browser, and rarely in the server's/site's configuration.

Or maybe say that while they're actionable for the Zulip developers, they're rarely actionable for a server administrator.

Reply:
Done.